anvio/__init__.py

```diff
@@ -2358,6 +2358,13 @@ def TABULATE(table, header, numalign="right", max_width=0):
              'help': "When declared, only reads that cover all positions will be reported. It is necessary to use this "
                      "flag if you want to perform oligotyping-like analyses on matching reads."}
                 ),
+    'add-coverage': (
```

Comment:
I wonder if one could use the existing `--init-gene-coverages` flag for this :)

Reply:
Technically, yes, we could. :) However, in order for the coverage/detection output to be customizable, it is necessary to initialize the profile database at the beginning of the run so that available headers to choose from are visible when using the flag `--list-available-output-headers`. If we were to use `--init-gene-coverages`, this initialization would load the coverage data, and that takes too much time if all you are doing is listing the headers.

(in fact, this initialization is what was causing the mysterious behavior that we couldn't figure out yesterday - I had two possible locations for profile db initialization - one at the beginning to handle the custom output headers case and one later in the code, and I placed `self.args.skip_create_gene_dbs = True` in the wrong one :)

Well, actually, now that I am thinking about it, I can simply turn `--init-gene-coverages` off if  `--list-available-output-headers` is on. 🤔 AND then this would eliminate the need for two functions that can initialize the profile database. Okay, never mind. :) I am convinced that this is the right way to go. I will change it.

Reply:
Well, it turns out that I will not be changing this, after all :) 
After some reflection, I think the design is better the way it is. The reason is that using `--init-gene-coverages` forces us to always initialize coverage data for all genes in the contigs database, even if not all of these genes have KOfam hits. (because we initialize the profile db with this parameter before we call `estimate_metabolism()` due to the need to know about available headers, as I alluded to in my previous comment). If we instead wait and call `profile_db.init_gene_level_coverage_stats_dicts()` after we know which genes we care about, it prevents us from processing this information for irrelevant gene calls, which should save us both time and memory. 

@meren, if you read this and still think it is better to go with `--init-gene-coverages` rather than `--add-coverage`, please let me know :)

Reply:
I am reading EVERYTHING and I am not thinking it is better to go with `--init-gene-coverages` :) apologies for the extra work and thanks for considering it. I agree that there is way too much power packed behind that community parameter in `dbops` and we can't just throw it around like that.